src/sk_gen.rs

```diff
@@ -11,13 +11,11 @@ pub(crate) fn genpoly_gen(out: &mut [Gf; SYS_T], f: &[Gf; SYS_T]) -> isize {
 
     mat[0][1..SYS_T].fill(0);
 
-    for i in 0..SYS_T {
-        mat[1][i] = f[i];
-    }
+    mat[1][..SYS_T].copy_from_slice(&f[..SYS_T]);
```

Comment:
Same here.. Any reason this could not just be `mat[1] = *f;`?

Reply:
I still had the c-style syntax as to how to write for loops in mind when initially porting this to rust,
also I learned the rust language quite recently so I wasn't fully aware of this simple syntax :)